src/components/actions/helpers.ts

```diff
@@ -0,0 +1,30 @@
+import { ActionInTemplate } from '../../types'
+import DBConnect from '../databaseConnect'
+
+export const swapOutAliasedActions = async (templateId: number, action: ActionInTemplate) => {
+  // Fetch aliased action from database
+  const {
+    condition,
+    parameter_queries: { code, condition: altCondition, ...overrideParams },
+  } = action
+  if (!code) throw new Error('Missing code in aliased action')
+
+  const aliasedAction = await DBConnect.getSingleTemplateAction(templateId, code)
+
+  if (!aliasedAction) throw new Error('No Action matching alias')
+
+  // Override condition if specified
```

Comment:
Hmm, I think a boolean would be more explanatory. `shouldOverrideCondition` as a parameter, would signify that we should use the `filed` condition of alias instead of the aliased one.

Reply:
I dunno, much of the time it's better to leave out the condition and let the original action's condition be considered.
Then if you want to change it, just provide a new condition parameter -- I don't see why you'd want to also have to provide a boolean flag as well. The condition parameter only ever exists when `shouldOverrideCondition` would be `true` -- it's implicitly true.

My thinking was that these parameters all work like when spreading objects and having overrides. Like:

```
overrides = {option1: true, option2: "A"}

baseSettings = {option1: false, option2: "B", option3: 99, option4: "happy"}
```
Then `activeSettings = {...baseSettings, ...overrides}`

I use this pattern a lot, it seems nice and self-explanatory to me. I don't like a bunch of extraneous compulsory, hard-to-remember flags unless they're unavoidable.

I agree that I could be clearer in the docs though -- I'll try and improve the explanation.

Reply:
I see your confusion probably came from this:
> To override the condition, you can either set the "condition" field directly, or supply a "condition" parameter (which takes priority). The one problem with the native "condition" field, though, is that it is always `true` by default, so we wouldn't want to always override with that. Therefore, the "condition" field only overrides the aliased actions condition if it is something other than `true`. (If you want it to be `true` (i.e. always run, then set `condition: true` in parameters))

Okay, I think I can see the point of the "shouldOverrideCondition" parameter -- but you should only need it in the rare situation where the alias has `condition=true` AND you want that to over-ride the aliased action.

So to summarise:
- default behaviour: original (aliased) action condition takes priority
- if the alias has a condition that is anything other than `true`, then *that* overrides the original action
- if the alias has `condition=TRUE` AND we *want* this to take priority, then we need to add the `shouldOverrideCondition` flag as a parameter, but we shouldn't need it at all in those other two situations.

Sound okay?

(Another way would be to make the default condition `null` rather than `true`, but I think that creates more problems than it solves 😆)

Reply:
Okay, I've updated the docs to reflect the above: 5b22f6fc

See if that makes sense and, if so, I'll update the actual code to reflect it.

Reply:
It's a simple code change: 4fbe57d8

I might have to update my snapshot for "PreviewAction" to reflect this, I'll check before you get to it ;)

Reply:
yeah, sorry if my PR reviews are a bit late and quite a few upstream changed need to be done because of it, i've pmed saying it's probably best to just do the changes to the latest pr in the stack to avoid conflicts etc..